src/morphology_workflows/repair.py

```diff
@@ -384,8 +372,44 @@ def _create_db_row(_data, zero_diameter_path, unravel_path, repair_path, extensi
     return name, data, m
 
 
-def make_release(df, _, zero_diameter_path, unravel_path, repair_path, extensions):
+def _set_layer(df):
+    """Set layer values from mtype name if no layer column is present."""
+    if "layer" not in df.columns:
+        for gid in df.index:
+            mtype = df.loc[gid, "mtype"]
+            if isinstance(mtype, str):
+                if len(mtype) > 1:
+                    try:
+                        layer = int(mtype[1])
+                    except ValueError:
+                        layer = "no_layer"
+            else:
+                layer = "no_mtype"
+
+            df.loc[gid, "layer"] = layer
+            df.loc[gid, "mtype"] = mtype
+
+
+def _duplicate_layers(df):
+    """Duplicate entries if layer name has mixed layers, i.e. L23_PC."""
+    mtypes = df.mtype.unique()
+    for mtype in mtypes:
+        layer = mtype.split("_")[1:]
+        if len(layer) > 1:
+            _df = df[df.mtype == mtype]
+            _df.layer = layer[1]
+            df = pd.concat([df, _df])
```

Comment:
This supposes that there is either 1 or 2 layers in the name, but is it possible to have more than 2?

Reply:
And as far as I can see, this will create entries with `mtype == '3' and`  keep entries with `mtype == 'L23_PC'`. Shouldn't we create entries with `mtype == 'L2_PC'` and `mtype == 'L3_PC'` instead?

Reply:
yes, mtype should be the same, just duplicating rows for double layers (they are always adjacent layers)